### PR TITLE
Support for "tag" data on Polyface

### PR DESCRIPTION
**Pull request by EarlinLutz**
* TaggedGeometryData class to hold application-specific markup on Polyface.
* IndexedPolyface has an optional member "taggedGeometry" which is an array of TaggedGeometryData.
* Specific first use is to identify the polyface as the crude mesh of a subdivision surface
* TaggedGeometryData has 2 integer tags `tagA` and `tagB`.
   * For subdivision mesh, `tagA` value is `TaggedDataConstants.TaggedGeometryDataTagType.SubdivisionSurface`
   * This is further qualified by `tagB` value which is one of:
      * `TaggedGeometryConstants.SubdivisionMethod.CatumullClark`
      * `TaggedGeometryConstants.SubdivisionMethod.Loop`
      * `TaggedGeometryConstants.SubdivisionMethod.DooSabin`
      * `TaggedGeometryConstants.SubdivisionMethod.ChooseBasedOnFacets`
      * These codes affect display (via imodel02 changes that are in process)
  * Support changes include
    * IMJS json schema read and write
    * BGFB flatbuffer read and write
 * Incidental changes in AustralianRailCorpSpiral tests to improve coverage.

### CODE REVIEW

**Review by mgooding — APPROVED** [review]

File: core/geometry/src/serialization/BGFBAccessors.ts [hunk 1]
```diff
@@ -4948,11 +5150,20 @@ expectedClosure(): number {
...
+taggedNumericData(obj?: TaggedNumericData): TaggedNumericData|null {
+  const offset = this.bb!.__offset(this.bb_pos, 40);
+  return offset ? (obj || new TaggedNumericData()).__init(this.bb!.__indirect(this.bb_pos + offset), this.bb!) : null;
+}
+
 /**
  * @param flatbuffers.Builder builder
  */
 static startPolyface(builder: flatbuffers.Builder) {
-  builder.startObject(18);
```
Comment (bbastings) [inline core/geometry/src/serialization/BGFBAccessors.ts:4955]: What's this hard-code 18/19 about?

File: core/geometry/src/geometry3d/Matrix3d.ts [hunk 2]
```diff
@@ -294,10 +294,45 @@ export class Matrix3d implements BeJSONFunctions {
    * @param tol optional tolerance for comparisons by Geometry.isDistanceWithinTol
    */
   public isAlmostEqual(other: Matrix3d, tol?: number): boolean {
-    if (tol)
       return Geometry.isDistanceWithinTol(this.maxDiff(other), tol);
-    return Geometry.isSmallMetricDistance(this.maxDiff(other));
   }
+  /** Test if this Matrix3d and have almost equal Z column and have X and Y columns differing only by a rotation around that Z.
```
Comment (bbastings) [inline core/geometry/src/geometry3d/Matrix3d.ts:299]: Description could use some work...

File: core/geometry/src/polyface/TaggedNumericData.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+*--------------------------------------------------------------------------------------------*/
+
+import { Geometry } from "../Geometry";
+
+/** @packageDocumentation
+ * @module Polyface
+ */
+
+/**
+ * `TaggedNumericConstants` defines enums with constant values for use in tags of `TaggedNumericData`
```
Comment (pmconne) [inline core/geometry/src/polyface/TaggedNumericData.ts:13]: Can you use doc links here and elsewhere so people don't have to search to find out what "TaggedNumericData" is?
`[[TaggedNumericData]]`

File: core/geometry/src/polyface/TaggedNumericData.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    FixedDepth = -100,
+    /** pair (FixedDepth, index) indicates absolute tolerance with value in doubleData[index] */
+    AbsoluteTolerance = -101,
+    /** pair (FixedDepth, index) indicates tolerance as a fraction of base mesh range is found in doubleData[index] */
+    FractionOfRangeBoxTolerance = -102
+  }
+}
+/**
+ * Structure with 2 integer tags and optional arrays of integers, doubles, points, vectors, and geometry.
+ * * In typescript/javascript, all integer numbers that can be non-integer.  Please do not insert non-integers in the integer array.
```
Comment (pmconne) [inline core/geometry/src/polyface/TaggedNumericData.ts:50]: Did you consider using growable Int32/Float64Arrays instead?

File: core/geometry/src/polyface/TaggedNumericData.ts (hunk 3 above)
Comment (EarlinLutz) [reply core/geometry/src/polyface/TaggedNumericData.ts]: Ah, that's easy.  Tx.

**Review by bbastings — APPROVED** [review]

File: core/geometry/src/serialization/BGFBAccessors.ts (hunk 1 above)
Comment (EarlinLutz) [reply core/geometry/src/serialization/BGFBAccessors.ts:4955]: This is machine generated flatbuffer access codes with offsets.

**Pull request merged**